datafusion/physical-optimizer/src/filter_pushdown.rs

```diff
@@ -362,17 +363,25 @@ use itertools::izip;
 /// [`ProjectionExec`]: datafusion_physical_plan::projection::ProjectionExec
 /// [`AggregateExec`]: datafusion_physical_plan::aggregates::AggregateExec
 #[derive(Debug)]
-pub struct FilterPushdown {}
+pub struct FilterPushdown {
+    phase: FilterPushdownPhase,
+    name: String,
+}
 
 impl FilterPushdown {
-    pub fn new() -> Self {
-        Self {}
+    fn new(phase: FilterPushdownPhase) -> Self {
+        Self {
+            phase,
+            name: format!("FilterPushdown({phase})"),
```

Comment:
Perhaps we could keep formatting the original rule as `FilterPushdown` and `FilterPushdown(Dynamic)`?

Reply:
I like `FilterPushdown` and `FilterPushdown(Dynamic)`

Reply:
I feel that if we do that we should rename the enum to `Static/Dynamic`. I think it's confusing to mix `Static/Dynamic` and `Pre/Post`.

I like `Pre/Post` because that's the actual reason we're splitting it into two phases: one runs before other optimizations, one runs after.